Compilation error on macOS 10.15: `fatal error: 'Virtualization/Virtualization.h' file not found`

```
$ make
rm -rf _output vendor
mkdir -p _output/bin
cp -a ./cmd/lima _output/bin/lima
# The hostagent must be compiled with CGO_ENABLED=1 so that net.LookupIP() in the DNS server
# calls the native resolver library and not the simplistic version in the Go library.
CGO_ENABLED=1 go build -ldflags="-s -w -X github.com/lima-vm/lima/pkg/version.Version=199216f" -o _output/bin/limactl ./cmd/limactl
go: downloading github.com/AlecAivazis/survey/v2 v2.3.6
go: downloading github.com/alessio/shellescape v1.[4](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:5).1
go: downloading github.com/containerd/containerd v1.6.10
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/docker/go-units v0.[5](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:6).0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/mattn/go-isatty v0.0.1[6](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:7)
go: downloading github.com/mattn/go-shellwords v1.0.12
go: downloading github.com/sirupsen/logrus v1.9.0
go: downloading github.com/spf13/cobra v1.6.1
go: downloading github.com/kballard/go-shellquote v0.0.0-2018042803000[7](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:8)-95032a[8](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:9)2bc51
go: downloading golang.org/x/term v0.1.0
go: downloading github.com/norouter/norouter v0.6.4
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/lima-vm/sshocker v0.3.0
go: downloading github.com/miekg/dns v1.1.50
go: downloading github.com/nxadm/tail v1.4.8
go: downloading github.com/goccy/go-yaml v1.[9](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:10).6
go: downloading github.com/opencontainers/go-digest v1.0.0
go: downloading github.com/xorcare/pointer v1.2.2
go: downloading github.com/containers/gvisor-tap-vsock v0.4.1-0.20220920072955-5b1aff8ba743
go: downloading golang.org/x/sync v0.1.0
go: downloading github.com/digitalocean/go-qemu v0.0.0-202[10](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:11)326154740-ac9e0b687001
go: downloading github.com/cyphar/filepath-securejoin v0.2.3
go: downloading golang.org/x/sys v0.2.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
go: downloading golang.org/x/text v0.4.0
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/pkg/sftp v1.13.4
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: downloading golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2
go: downloading golang.org/x/net v0.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading gvisor.dev/gvisor v0.0.0-20220908032458-edc830a43ba6
go: downloading inet.af/tcpproxy v0.0.0-20220326234310-be3ee21c9fa0
go: downloading github.com/digitalocean/go-libvirt v0.0.0-20201209184759-e2a69bcd5bd1
go: downloading github.com/diskfs/go-diskfs v1.2.0
go: downloading github.com/cheggaaa/pb/v3 v3.1.0
go: downloading github.com/containerd/continuity v0.3.0
go: downloading github.com/mattn/go-colorable v0.1.12
go: downloading github.com/Code-Hex/vz/v3 v3.0.0
go: downloading github.com/kr/fs v0.1.0
go: downloading golang.org/x/crypto v0.1.0
go: downloading github.com/fsnotify/fsnotify v1.5.1
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/insomniacslk/dhcp v0.0.0-20220504074936-1ca156eafb9f
go: downloading github.com/apparentlymart/go-cidr v1.1.0
go: downloading github.com/google/gopacket v1.1.19
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading github.com/google/btree v1.0.1
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/mattn/go-runewidth v0.0.12
go: downloading golang.org/x/mod v0.6.0
go: downloading google.golang.org/grpc v1.47.0
go: downloading github.com/u-root/uio v0.0.0-20210528[11](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:12)4334-82958018845c
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading gopkg.in/djherbis/times.v1 v1.2.0
go: downloading google.golang.org/genproto v0.0.0-20220502173005-c8bf987b8c21
go: downloading github.com/golang/protobuf v1.5.2
go: downloading google.golang.org/protobuf v1.28.0
# github.com/Code-Hex/vz/v3
In file included from ../../../go/pkg/mod/github.com/!code-!hex/vz/v3@v3.0.0/audio.go:6:
In file included from ./virtualization_11.h:9:
./virtualization_helper.h:25:9: warning: macOS [12](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:13).3 API has been disabled [-W#pragma-messages]
#pragma message("macOS 12.3 API has been disabled")
        ^
./virtualization_helper.h:32:9: warning: macOS [13](https://github.com/lima-vm/lima/actions/runs/3494156702/jobs/5849687766#step:4:14) API has been disabled [-W#pragma-messages]
#pragma message("macOS 13 API has been disabled")
        ^
In file included from ../../../go/pkg/mod/github.com/!code-!hex/vz/v3@v3.0.0/audio.go:6:
./virtualization_11.h:10:9: fatal error: 'Virtualization/Virtualization.h' file not found
#import <Virtualization/Virtualization.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings and 1 error generated.
make: *** [_output/bin/limactl] Error 2
Error: Process completed with exit code 2.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -145,6 +145,23 @@ jobs:
     # GHA macOS is slow and flaky, so we only test a few YAMLS here.
     # Other yamls are tested on Linux instances of Cirrus.
 
+  catalina:
+    name: "macOS 10.15 (deprecated)"
+    # Will be fully unsupported by 12/1/22 https://github.com/actions/runner-images/issues/5583
+    runs-on: macos-10.15
+    timeout-minutes: 20
+    steps:
+    - uses: actions/setup-go@v3
+      with:
+        go-version: 1.19.x
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 1
+    - name: Make
+      run: make
+    - name: Install
+      run: make install
+
   vmnet:
     name: "VMNet test"
     runs-on: macos-11

--- a/Makefile
+++ b/Makefile
@@ -13,12 +13,21 @@ bat = .bat
 exe = .exe
 endif
 
+GO_BUILDTAGS ?=
+ifeq ($(GOOS),darwin)
+MACOS_VERSION=$(shell sw_vers -productVersion | cut -d . -f 1)
+ifeq ($(shell test $(MACOS_VERSION) -lt 13; echo $$?),0)
+# The "vz" mode needs macOS 13 or later
+GO_BUILDTAGS += no_vz
+endif
+endif
+
 PACKAGE := github.com/lima-vm/lima
 
 VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 VERSION_TRIMMED := $(VERSION:v%=%)
 
-GO_BUILD := $(GO) build -ldflags="-s -w -X $(PACKAGE)/pkg/version.Version=$(VERSION)"
+GO_BUILD := $(GO) build -ldflags="-s -w -X $(PACKAGE)/pkg/version.Version=$(VERSION)" -tags "$(GO_BUILDTAGS)"
 
 .NOTPARALLEL:
 

--- a/pkg/vz/vm_darwin.go
+++ b/pkg/vz/vm_darwin.go
@@ -1,3 +1,6 @@
+//go:build darwin && !no_vz
+// +build darwin,!no_vz
+
 package vz
 
 import (

--- a/pkg/vz/vz_driver_darwin.go
+++ b/pkg/vz/vz_driver_darwin.go
@@ -1,3 +1,6 @@
+//go:build darwin && !no_vz
+// +build darwin,!no_vz
+
 package vz
 
 import (

--- a/pkg/vz/vz_driver_others.go
+++ b/pkg/vz/vz_driver_others.go
@@ -1,14 +1,16 @@
-//go:build !darwin
-// +build !darwin
+//go:build !darwin || no_vz
+// +build !darwin no_vz
 
 package vz
 
 import (
 	"context"
-	"fmt"
+	"errors"
 
 	"github.com/lima-vm/lima/pkg/driver"
 )
+
+var ErrUnsupported = errors.New("vm driver 'vz' needs macOS 13 or later (Hint: try recompiling Lima if you are seeing this error on macOS 13)")
 
 type LimaVzDriver struct {
 	*driver.BaseDriver
@@ -21,17 +23,17 @@ func New(driver *driver.BaseDriver) *LimaVzDriver {
 }
 
 func (l *LimaVzDriver) Validate() error {
-	return fmt.Errorf("driver 'vz' is only supported on darwin")
+	return ErrUnsupported
 }
 
 func (l *LimaVzDriver) CreateDisk() error {
-	return fmt.Errorf("driver 'vz' is only supported on darwin")
+	return ErrUnsupported
 }
 
 func (l *LimaVzDriver) Start(ctx context.Context) (chan error, error) {
-	return nil, fmt.Errorf("driver 'vz' is only supported on darwin")
+	return nil, ErrUnsupported
 }
 
 func (l *LimaVzDriver) Stop(_ context.Context) error {
-	return fmt.Errorf("driver 'vz' is only supported on darwin")
+	return ErrUnsupported
 }

